src/type/definition.js

```diff
@@ -294,13 +294,27 @@ function resolveThunk<T>(thunk: Thunk<T>): T {
 export class GraphQLScalarType {
   name: string;
   description: ?string;
+  ofType: ?GraphQLScalarType;
 
   _scalarConfig: GraphQLScalarTypeConfig<*, *>;
 
   constructor(config: GraphQLScalarTypeConfig<*, *>): void {
     assertValidName(config.name);
     this.name = config.name;
     this.description = config.description;
+    this.ofType = config.ofType || null;
+    if (this.ofType) {
+      const ofTypeName = this.ofType.name;
+      invariant(
+        ofTypeName === 'String' ||
+        ofTypeName === 'Int' ||
+        ofTypeName === 'Float' ||
+        ofTypeName === 'Boolean' ||
+        ofTypeName === 'ID',
```

Comment:
@leebyron `graphql-js` already have the same list in multiple places + we have the same code in all of our GraphQL projects. It would be great if you review #924 which adds `isBuiltInScalar` predicate.